cmake/macros/CompositeSources.cmake

```diff
@@ -47,35 +44,69 @@ function(composite_sources target sources_var)
     return()
   endif()
 
+  # Get rid of the periods in the extensions and create each list of composite
+  # sources.
+  set(extensions "")
+  foreach(extension ${COMPOSITE_SOURCE_EXTENSIONS})
+    string(MAKE_C_IDENTIFIER extension ${extension})
+    list(APPEND extensions ${extension})
+    set(sources_${extension} "")
+  endforeach()
+
+  # Sort each source file into a list.
+  foreach(source ${sources})
+    get_filename_component(extension "${source}" EXT)
+    string(MAKE_C_IDENTIFIER extension ${extension})
+    list(APPEND sources_${extension} "${source}")
+  endforeach()
+
+  set(sorted_sources "")
+  foreach(extension ${extensions})
```

Comment:
Come to think of it, instead of discovering `extensions` dynamically, how about just iterating `COMPOSITE_SOURCE_EXTENSIONS`?